pom.xml

```diff
@@ -3081,5 +3094,17 @@
     <profile>
       <id>sparkr</id>
     </profile>
+    <profile>
+      <id>aarch64</id>
+      <properties>
+        <leveldbjni.group>org.openlabtesting.leveldbjni</leveldbjni.group>
+      </properties>
+      <activation>
+        <os>
+          <family>linux</family>
+          <arch>aarch64</arch>
```

Comment:
Fisrt, unforturnately, the main repo seems is not under maintaining, I've asked committers to public a new release which supports aarch64 platform, but there is no reply. Second, if the main repo have 1.9 version, I believe the 1.9 version should support aarch64 platform, then I think we can migrate to use it directly? And last, if you want to use org.openlabtesting.leveldbjni:leveldbjni-all.1.9  I will make it :)

About the 'activation' I think it only specify the default profile depends on the build machine, you can activete with -Paarch64 on any other build machine.

Reply:
Actually, you've had prompt responses repeatedly: https://mail-archives.apache.org/mod_mbox/spark-dev/201911.mbox/%3CCAEccTyxQ31BVP3vty0h_VxE_%3DfE7OWUg%3DqpumuZuK5kDys21JQ%40mail.gmail.com%3E I won't reply a fourth time, but here, you're suggesting there is still something that needs to be done to make aarch64 work. Why would anyone announce it works then? No, the project is not going to take on this goal for you; you are best placed to do this work. As before, you are getting prompt attention for your efforts.

I do not expect that Spark will ever release a separate aarch64 build. So, the ideal solution is to make the upstream version compatible with all architectures. I do not know what the main leveldbjni project is or isn't going to support; it's what I'm asking you. For now, relying on a fork under a profile could be OK. Are there any other differences in the fork?

Yes, you do not want to activate based on the build machine's architecture. It should be specified manually.

Reply:
@srowen, I am sorry didn't say it clear, 'I've asked committers to public a new release which supports aarch64 platform, but there is no reply.' ---- I said I've ask the org.fusesource.leveldbjni to public a new release(like 1.9) to support aarch64, not spark:)

Reply:
@srowen, and about the spark arm CI, now I will follow your suggestion only to send an email to user@ tell users the situation of the arm testing of spark.

Reply:
Oh I entirely misread that. Please disregard the first comment with my apologies.

Reply:
Are there any other differences in the fork?
--- @srowen, I fork fusesource/leveldbjni to theopenlab/leveldbjni, and yes, we've modified it to make it support aarch64, the details you can see https://github.com/theopenlab/leveldbjni/commit/aa6b14a50a9acb2681102975107784f052f1ceeb (the code refers to google/leveldb https://github.com/google/leveldb/commit/c4c38f9c1f3bb405fe22a79c5611438f91208d09#diff-b20d9097440605779c4984e3a1346085)
I only built new .so for aarch64 platform.

Reply:
fusesource/leveldbjni:leveldbjni-all.1.8  supports platforms: osx, linux32, linux64(x86), win32,win64, see https://mvnrepository.com/artifact/org.fusesource.leveldbjni/leveldbjni-all/1.8 
we add linux64-aarch64 to openlabtesting/leveldbjni:leveldbjni-all.1.8, see https://mvnrepository.com/artifact/org.openlabtesting.leveldbjni/leveldbjni-all/1.8

Reply:
@srowen, so is there possible migrate to use openlabtesting.leveldbjni:leveldbjni-all.1.8 directly in spark for all platforms supported?

Reply:
The most ideal outcome would be for the upstream project to adopt the change. It looks quite minor. Otherwise the fork might end up running behind upstream changes, but it sounds like there is not much upstream activity. Next-most ideal would be for Hadoop to adopt the same approach and we use a release that follows the same pattern. But the approach here is OK and maybe less intrusive as it won't affect the main build.

Reply:
https://issues.apache.org/jira/browse/HADOOP-16614 hadoop adopt the same approach, the code has been merged.